contracts/cosoul/SBT.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.2;
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract CoSoul is ERC721("CoSoul", "Soul"), Ownable {
+	using ECDSA for bytes32;
+
+	bool public initiated;
+	address public signer;
+	mapping(address => bool) public authorisedCallers;
+	mapping(uint256 => uint256) public transferNonces;
+	mapping(uint256 => uint256) public syncNonces;
+	mapping(uint256 => uint256) public burnNonces;
+	mapping(address => uint256) public mintNonces;
+	uint256 public counter;
+	mapping(uint256 => uint256) public blobs;
+	
+
+	modifier authorised(address _operator) {
+		require(authorisedCallers[_operator] || _operator == owner());
+		_;
+	}
+
+	function init(string memory __name, string memory __symbol, address _signer) external {
+		require(!initiated);
+		initiated = true;
+		_name = __name;
+		_symbol = __symbol;
+		_owner = msg.sender;
+		signer = _signer;
+	}
+
+	function setSigner(address _signer) external onlyOwner {
+		signer = _signer;
+	}
+
+
+	function setCallers(address _caller, bool _val) external onlyOwner {
+		authorisedCallers[_caller] = _val;
+	}
+
+	function getSlot(uint8 _slot, uint256 _tokenId) public view returns(uint256 value) {
```

Comment:
@OwlOfMoistness do you have some docs on how these work? I'd like to understand this a bit better.